test/end-to-end/duplication.test.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+var utils = require('./e2e-tools');
+var request = require('supertest');
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+var promise = protractor.promise;
+
+
+describe('test duplication of reports with different settings', function() {
+  before(utils.initDb);
+  after(utils.disconnectDropDb);
+
+  beforeEach(utils.resetDb);
+  beforeEach(utils.initAdmin.bind({}, 'asdfasdf'));
+  afterEach(utils.resetBrowser);
+  var reqParams = {
+  from: '9845098450',
+  fulltext: 'loremipsumdolor',
+  date: '2016-09-01',
+  keyword: 'test'
+  };
+
+  var chain = function() {
+    var defer = promise.defer();
+    defer.fulfill(true);
+    return defer.promise;
+  };
+  it('should listen with fetching:on and source:enabled', function(done) {
+    chain()
+    .then(function() {
+      utils.addSource('SMS GH', { nickname: 'hello', keywords: 'test' });
+    })
+    .then(function() {
+      utils.toggleFetching('On');
+    })
+    .then(function() {
+      utils.toggleSource('SMS GH', 'On');
+    })
+    .then(function() {
+      request('http://localhost:1111')
```

Comment:
make a function of the SMS Gh request. You can also abstract the 5 test, so you only pass the status of the toggles and the expected reports to a function, and don't duplicate so much code